// uac/lib/namespace.js
var ns = {
	Authorization: "http://ns.bergnet.org/uac/0.1/universal-access-control#Authorization",
	Filter: "http://ns.bergnet.org/uac/0.1/universal-access-control#Filter",
	ResourceAuthorization: "http://ns.bergnet.org/uac/0.1/universal-access-control#ResourceAuthorization",
	Read: "http://ns.bergnet.org/uac/0.1/universal-access-control#Read",
	Role: "http://ns.bergnet.org/uac/0.1/universal-access-control#Role",
	SimpleFilter: "http://ns.bergnet.org/uac/0.1/universal-access-control#SimpleFilter",
	TripleAuthorization: "http://ns.bergnet.org/uac/0.1/universal-access-control#TripleAuthorization",
	VariableFilter: "http://ns.bergnet.org/uac/0.1/universal-access-control#VariableFilter",
	Write: "http://ns.bergnet.org/uac/0.1/universal-access-control#Write",
	access: "http://ns.bergnet.org/uac/0.1/universal-access-control#access",
	agent: "http://ns.bergnet.org/uac/0.1/universal-access-control#agent",
	application: "http://ns.bergnet.org/uac/0.1/universal-access-control#application",
	children: "http://ns.bergnet.org/uac/0.1/universal-access-control#children",
	filter: "http://ns.bergnet.org/uac/0.1/universal-access-control#filter",
	graph: "http://ns.bergnet.org/uac/0.1/universal-access-control#graph",
	hasRole: "http://ns.bergnet.org/uac/0.1/universal-access-control#hasRole",
	mode: "http://ns.bergnet.org/uac/0.1/universal-access-control#mode",
	object: "http://ns.bergnet.org/uac/0.1/universal-access-control#object",
	predicate: "http://ns.bergnet.org/uac/0.1/universal-access-control#predicate",
	required: "http://ns.bergnet.org/uac/0.1/universal-access-control#required",
	statement: "http://ns.bergnet.org/uac/0.1/universal-access-control#statement",
	subject: "http://ns.bergnet.org/uac/0.1/universal-access-control#subject",
	value: "http://ns.bergnet.org/uac/0.1/universal-access-control#value",
	variable: "http://ns.bergnet.org/uac/0.1/universal-access-control#variable"
};


module.exports = ns;
